recipes-kernel/kernel-modules/kernel-module-isp-vvcam/0004-NAVQP-228-fix-VIDIOC_ENUMINPUT-for-imx219.patch

```diff
@@ -0,0 +1,21 @@
+From 1a0e1a9be248d9e9fd8dcdb72a15fe2b6684c0c7 Mon Sep 17 00:00:00 2001
+From: Alexander Yurtsev <alex@voxelbotics.com>
+Date: Thu, 19 Sep 2024 07:43:17 +0200
+Subject: [PATCH] NAVQP-228: fix VIDIOC_ENUMINPUT for imx219
+
+---
+ vvcam/v4l2/sensor/imx219/imx219_mipi_v3.c | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/vvcam/v4l2/sensor/imx219/imx219_mipi_v3.c b/vvcam/v4l2/sensor/imx219/imx219_mipi_v3.c
+index 2a2322a..74b8859 100644
+--- v4l2/sensor/imx219/imx219_mipi_v3.c
++++ v4l2/sensor/imx219/imx219_mipi_v3.c
+@@ -742,6 +742,7 @@ static long imx219_priv_ioctl(struct v4l2_subdev *sd,
+ 		ret= imx219_set_test_pattern(sensor, arg);
+ 		break;
+ 	default:
++		ret = -ENOTTY;
```

Comment:
Using default here extends the return of ENOTTY to other possible IOCTLs which looks excessive, and hypothetically can break other use cases where no ENOTTY for some IOCTLs were extended.
I suggest to provide a specific case label for ENUMINPUT IOCTL here.

Reply:
@dnbazhenov

I believe that returning -ENOTTY in the default label is a better solution, since this would filter out other ioctls that are not explicitly supported by the imx219 driver.

Note that the central v4l2 ioctl handler (__video_do_ioctl) also returns -ENOTTY if no ioctl ops structure is registered for a video device.